tools/windows-node-installer/pkg/cloudprovider/azure/vm.go

```diff
@@ -482,12 +494,144 @@ func (az *AzureProvider) constructStorageProfile(imageId string) (storageProfile
-	// We assume that BackendPoolID for worker vm follows the following format
-	// /subscriptions/$SUBSCRIPTID/resourceGroups/$INFRAID-rg/providers/Microsoft.Network/loadBalancers/$INFRAID/backendAddressPools/$INFRAID
-	var wbpi = fmt.Sprintf("/subscriptions/%s/resourceGroups/%[2]s-rg/providers/"+
-		"Microsoft.Network/loadBalancers/%[2]s/backendAddressPools/%[2]s", az.subscriptionID, az.infraID)
-	return &wbpi
+func (az *AzureProvider) getWorkerBackendAddressPools(ctx context.Context) ([]network.BackendAddressPool, error) {
+	// Create a filter to get a resource of type virtualMachine which contains 'worker' in its name
+	filter := "resourceType eq 'Microsoft.Compute/virtualMachines' and substringof('worker', name) and " +
+		"resourceGroup eq '" + az.resourceGroupName + "'"
+	// number of resources to retrieve
+	// https://learnk8s.io/kubernetes-node-size
+	// Kubernetes can technically accommodate 5000 nodes, considering that for optimal performance we limit it at 500 worker nodes
+	top := int32(500)
+	// Use getResource() function to get all the workerVMs by filtering VMs which have 'worker' in their name
+	// In the vmClient, they need full vmName. We can use list() but list would return all the VMs in the
+	// resource group, we will then have to filter it to get the worker VMs using go code iterating through
+	// all the VM names. getResource() does all that work for us. We just need to use the output in that case.
+	vmResources, err := az.getResource(ctx, filter, top)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get worker VMs: %v", err)
+	}
+	if vmResources == nil {
+		return nil, fmt.Errorf("failed to get worker VMs resource")
+	}
+
+	// keep track of workerVM load balancer
+	// using string as we are assuming all the worker VMs are behind a single load balancer
+	var backendAddressPool string
+
+	// Iterate over all the VM resources and get the worker backend pools attached to all the resources.
+	for _, vmResource := range vmResources {
+		if vmResource.ID == nil {
+			return nil, fmt.Errorf("id for VM resource cannot be nil")
+		}
+		vmName := extractResourceName(*vmResource.ID)
+		// get the backend address pool
+		workerVMBackendAddressPool, err := az.getVMBackendAddressPool(ctx, vmName)
+		if err != nil {
+			return nil, fmt.Errorf("unable to get backend address pools for vm %s", vmName)
+		}
+		if backendAddressPool == "" {
+			backendAddressPool = workerVMBackendAddressPool
+		} else if backendAddressPool != workerVMBackendAddressPool {
+			return nil, fmt.Errorf("multiple backend address pools for worker nodes is not supported")
+		}
+	}
+
+	// create an array of the backend address pools that are attached to worker nodes
+	// which will then be applied to the Windows node
+	lbBackendPoolAddressIDs := []network.BackendAddressPool{
+		{
+			ID: &backendAddressPool,
+		},
+	}
+
+	return lbBackendPoolAddressIDs, nil
+}
+
+// get the backend address pool associated with the VM
```

Comment:
Doc has to start with function name. You have done this in multiple places. Please fix.